packages/orbit-components/src/Timeline/TimelineStep/index.js

```diff
@@ -0,0 +1,160 @@
+// @flow
+import React from "react";
+import styled, { css } from "styled-components";
+
+import defaultTheme from "../../defaultTheme";
+import Badge from "../../Badge";
+import Text from "../../Text";
+import Stack from "../../Stack";
+import CheckCircle from "../../icons/CheckCircle";
+import Circle from "../../icons/CircleEmpty";
+import AlertCircle from "../../icons/AlertCircle";
+import CloseCircle from "../../icons/CloseCircle";
+import { useStatuses, useStep } from "../TimelineContext";
+import useMediaQuery from "../../hooks/useMediaQuery";
+
+import type { Props } from "./index";
+
+const IconWrapperStyled = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  min-width: ${({ theme }) => theme.orbit.spaceLarge};
+  min-height: ${({ theme }) => theme.orbit.spaceLarge};
+  svg {
+    background: ${({ theme }) => theme.orbit.paletteWhite};
+  }
+`;
+
+IconWrapperStyled.defaultProps = {
+  theme: defaultTheme,
+};
+
+const StatusIcon = ({ status }) => {
+  if (status === "success") return <CheckCircle color="success" />;
+  if (status === "warning") return <AlertCircle color="warning" />;
+  if (status === "critical") return <CloseCircle color="critical" />;
+
+  return <Circle color="tertiary" size="small" />;
+};
+
+const renderStatus = (status, theme) => {
+  if (status === "success") return theme.orbit.colorTextSuccess;
+  if (status === "warning") return theme.orbit.colorTextWarning;
+  if (status === "critical") return theme.orbit.colorTextCritical;
+
+  return theme.orbit.paletteCloudNormalHover;
+};
+
+const ProgressLineStyled = styled.span`
+  ${({ desktop, theme, status, isLeft }) => css`
+    ${!desktop && `position: absolute`};
+    width: ${desktop ? "50%" : "2px"};
+    ${isLeft ? `left: 0` : `right: -20px`};
+    z-index: -1;
```

Comment:
I would not prefer to use negative z-index here. It will cause a trouble on IE 10/11 and because of it, you're forced to use `useModalContext`, especially `isInsideModal` boolean. In the end it will add unnecessary complexity that will make future refactoring of modal related components harder and we will have to consider this also in z-index framework.

What I suggest is to get rid off negative z-index, the usage of `isInsideModal` and rather to use `relative` position on the `icon` or its wrapper. We can also tweak the height, since we're using `calc` already, so it will be fine.

There will be probably a small problem with the usage of `Circle` icon, that has different inner size, but it can be tweak with other interpolations – render functions.